src/vmod_vsthrottle.c

```diff
@@ -241,7 +241,7 @@ run_gc(double now, unsigned part)
 	/* XXX: Assert mtx is held ... */
 	VRBT_FOREACH_SAFE(x, tbtree, buckets, y) {
 		CHECK_OBJ_NOTNULL(x, TBUCKET_MAGIC);
-		if (now - x->last_used > x->period) {
+		if (now - x->last_used > (x->period + x->block)) {
```

Comment:
I think this is a bit off. 

``x->block`` is monotonic abstime, so what you're doing here is comparing a time difference to a timestamp. I think the logic in #207 had the right idea.

Reply:
ah, thank you @daghf, you are right as usual